2023-09-18 17:49:04.502552: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2023-09-18 17:49:04.502712: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2023-09-18 17:49:12,400 | app.py:151 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2023-09-18 17:49:12,493 | app.py:172 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2023-09-18 17:49:12,493 | server.py:86 | Initializing global parameters
INFO flwr 2023-09-18 17:49:12,493 | server.py:273 | Requesting initial parameters from one random client
INFO flwr 2023-09-18 17:49:12,593 | server.py:277 | Received initial parameters from one random client
INFO flwr 2023-09-18 17:49:12,593 | server.py:88 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:49:12,609 | server.py:95 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.4033333333333333}
INFO flwr 2023-09-18 17:49:12,609 | server.py:101 | FL starting
DEBUG flwr 2023-09-18 17:49:12,856 | server.py:223 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2023-09-18 17:49:12,878 | server.py:237 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2023-09-18 17:49:12,880 | fedavg.py:243 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:49:12,888 | server.py:125 | fit progress: (1, 0.3879566118399436, {'accuracy': 0.8233333333333334}, 0.2783824999999993)
DEBUG flwr 2023-09-18 17:49:12,888 | server.py:173 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:49:12,915 | server.py:187 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2023-09-18 17:49:12,916 | fedavg.py:274 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2023-09-18 17:49:12,916 | server.py:223 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:49:12,935 | server.py:237 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:49:12,944 | server.py:125 | fit progress: (2, 0.3833553933993886, {'accuracy': 0.8333333333333334}, 0.33533289999999916)
DEBUG flwr 2023-09-18 17:49:12,944 | server.py:173 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:49:12,964 | server.py:187 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2023-09-18 17:49:12,965 | server.py:223 | fit_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:49:12,984 | server.py:237 | fit_round 3 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:49:12,995 | server.py:125 | fit progress: (3, 0.3833263077345959, {'accuracy': 0.8333333333333334}, 0.3858908000000003)
DEBUG flwr 2023-09-18 17:49:12,995 | server.py:173 | evaluate_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:49:13,012 | server.py:187 | evaluate_round 3 received 3 results and 0 failures
DEBUG flwr 2023-09-18 17:49:13,012 | server.py:223 | fit_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:49:13,043 | server.py:237 | fit_round 4 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:49:13,055 | server.py:125 | fit progress: (4, 0.38249626485748384, {'accuracy': 0.8233333333333334}, 0.4455671999999993)
DEBUG flwr 2023-09-18 17:49:13,055 | server.py:173 | evaluate_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:49:13,082 | server.py:187 | evaluate_round 4 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:49:13,083 | server.py:223 | fit_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:49:13,114 | server.py:237 | fit_round 5 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:49:13,126 | server.py:125 | fit progress: (5, 0.38250338005976453, {'accuracy': 0.8233333333333334}, 0.5167082999999995)
DEBUG flwr 2023-09-18 17:49:13,127 | server.py:173 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:49:13,149 | server.py:187 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:49:13,149 | server.py:223 | fit_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:49:13,178 | server.py:237 | fit_round 6 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:49:13,190 | server.py:125 | fit progress: (6, 0.38250318353862217, {'accuracy': 0.8233333333333334}, 0.5805883000000005)
DEBUG flwr 2023-09-18 17:49:13,190 | server.py:173 | evaluate_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:49:13,211 | server.py:187 | evaluate_round 6 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:49:13,211 | server.py:223 | fit_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:49:13,239 | server.py:237 | fit_round 7 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:49:13,251 | server.py:125 | fit progress: (7, 0.38250318364290914, {'accuracy': 0.8233333333333334}, 0.6415561000000007)
DEBUG flwr 2023-09-18 17:49:13,251 | server.py:173 | evaluate_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:49:13,276 | server.py:187 | evaluate_round 7 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:49:13,277 | server.py:223 | fit_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:49:13,307 | server.py:237 | fit_round 8 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:49:13,320 | server.py:125 | fit progress: (8, 0.38250318357496393, {'accuracy': 0.8233333333333334}, 0.7100472)
DEBUG flwr 2023-09-18 17:49:13,320 | server.py:173 | evaluate_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:49:13,347 | server.py:187 | evaluate_round 8 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:49:13,347 | server.py:223 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:49:13,387 | server.py:237 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:49:13,403 | server.py:125 | fit progress: (9, 0.3830771633283049, {'accuracy': 0.82}, 0.7931796000000002)
DEBUG flwr 2023-09-18 17:49:13,403 | server.py:173 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:49:13,439 | server.py:187 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:49:13,439 | server.py:223 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:49:13,478 | server.py:237 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:49:13,496 | server.py:125 | fit progress: (10, 0.383077860035559, {'accuracy': 0.82}, 0.8870240000000003)
DEBUG flwr 2023-09-18 17:49:13,497 | server.py:173 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:49:13,536 | server.py:187 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2023-09-18 17:49:13,536 | server.py:147 | FL finished in 0.9267637999999998
INFO flwr 2023-09-18 17:49:13,537 | app.py:218 | app_fit: losses_distributed [(1, 0.4008208904415369), (2, 0.40032600425183773), (3, 0.40032497607171535), (4, 0.4129248382080169), (5, 0.4129245969511214), (6, 0.41292454302310944), (7, 0.41292454302310944), (8, 0.41292454302310944), (9, 0.4143080157893045), (10, 0.4143078399556024)]
INFO flwr 2023-09-18 17:49:13,537 | app.py:219 | app_fit: metrics_distributed_fit {}
INFO flwr 2023-09-18 17:49:13,537 | app.py:220 | app_fit: metrics_distributed {}
INFO flwr 2023-09-18 17:49:13,538 | app.py:221 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.3879566118399436), (2, 0.3833553933993886), (3, 0.3833263077345959), (4, 0.38249626485748384), (5, 0.38250338005976453), (6, 0.38250318353862217), (7, 0.38250318364290914), (8, 0.38250318357496393), (9, 0.3830771633283049), (10, 0.383077860035559)]
INFO flwr 2023-09-18 17:49:13,538 | app.py:222 | app_fit: metrics_centralized {'accuracy': [(0, 0.4033333333333333), (1, 0.8233333333333334), (2, 0.8333333333333334), (3, 0.8333333333333334), (4, 0.8233333333333334), (5, 0.8233333333333334), (6, 0.8233333333333334), (7, 0.8233333333333334), (8, 0.8233333333333334), (9, 0.82), (10, 0.82)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.99950817,  1.11871534,  0.73411183,  0.52231315,  0.10561787,
         0.0843088 ,  0.0152789 , -0.08767055]]), array([0.03265616])]
[array([[-1.99950817,  1.11871534,  0.73411183,  0.52231315,  0.10561787,
         0.0843088 ,  0.0152789 , -0.08767055]]), array([0.03265616])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.97997429,  1.06169048,  0.75004435,  0.50306703,  0.16838304,
         0.07361041, -0.00352112, -0.10182241]]), array([0.07418736])]
[array([[-1.97997429,  1.06169048,  0.75004435,  0.50306703,  0.16838304,
         0.07361041, -0.00352112, -0.10182241]]), array([0.07418736])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98040078,  1.06140724,  0.75004624,  0.5026163 ,  0.16884135,
         0.07349211, -0.00388104, -0.10192152]]), array([0.07551921])]
[array([[-1.98040078,  1.06140724,  0.75004624,  0.5026163 ,  0.16884135,
         0.07349211, -0.00388104, -0.10192152]]), array([0.07551921])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.88583021,  0.94914575,  0.76362189,  0.43045687,  0.19903081,
         0.08944189, -0.02619765,  0.01521965]]), array([-0.04037159])]
[array([[-1.88583021,  0.94914575,  0.76362189,  0.43045687,  0.19903081,
         0.08944189, -0.02619765,  0.01521965]]), array([-0.04037159])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.88547593,  0.9491573 ,  0.76399635,  0.43055591,  0.19898668,
         0.08930343, -0.02598444,  0.01553466]]), array([-0.04177271])]
[array([[-1.88547593,  0.9491573 ,  0.76399635,  0.43055591,  0.19898668,
         0.08930343, -0.02598444,  0.01553466]]), array([-0.04177271])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.88549594,  0.94916814,  0.76400567,  0.43056009,  0.19899167,
         0.08929868, -0.0259854 ,  0.01553708]]), array([-0.04177699])]
[array([[-1.88549594,  0.94916814,  0.76400567,  0.43056009,  0.19899167,
         0.08929868, -0.0259854 ,  0.01553708]]), array([-0.04177699])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.88549609,  0.94916828,  0.76400582,  0.43056016,  0.19899174,
         0.08929871, -0.02598535,  0.01553716]]), array([-0.04177744])]
[array([[-1.88549609,  0.94916828,  0.76400582,  0.43056016,  0.19899174,
         0.08929871, -0.02598535,  0.01553716]]), array([-0.04177744])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.8854961 ,  0.94916828,  0.76400583,  0.43056017,  0.19899174,
         0.08929871, -0.02598535,  0.01553716]]), array([-0.04177744])]
[array([[-1.8854961 ,  0.94916828,  0.76400583,  0.43056017,  0.19899174,
         0.08929871, -0.02598535,  0.01553716]]), array([-0.04177744])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.93160182,  0.95098602,  0.76997848,  0.45950294,  0.20144845,
         0.08022424, -0.03407425,  0.02278918]]), array([-0.055843])]
[array([[-1.93160182,  0.95098602,  0.76997848,  0.45950294,  0.20144845,
         0.08022424, -0.03407425,  0.02278918]]), array([-0.055843])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.93165023,  0.95104906,  0.76996736,  0.45982441,  0.20150382,
         0.08029758, -0.03408917,  0.02285796]]), array([-0.05628008])]
[array([[-1.93165023,  0.95104906,  0.76996736,  0.45982441,  0.20150382,
         0.08029758, -0.03408917,  0.02285796]]), array([-0.05628008])]
>>>>>>>>>>>>>run time:
0.453125s
